lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::PersistentTransmitter.batch do |transmitter|
+  #         transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #     end  
+
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+   
+    def initialize
+      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
+    end
+
+    def send_message!(payload, routing_key, type: nil, headers: nil, message_id: :auto_generate)
```

Comment:
I wonder if this should re-use the other method in `Transmitter`?  We should use the existing error handling methods - if we use this class to send a ton of messages and it blows up halfway through it will be hard to recover from.  It might also be surprising to devs expecting those messages to get sent async and thus make it hard to broadly adopt this.

I'm wondering if we make the other method like this:

```ruby
 def send_message!(payload, routing_key, type: nil, headers: nil, message_id: :auto_generate, close_connector: true)
      publish_options = Pwwka::PublishOptions.new(
        routing_key: routing_key,
        message_id: message_id,
        type: type,
        headers: headers
      )
      logf "START Transmitting Message on id[%{id}] %{routing_key} -> %{payload}", id: publish_options.message_id, routing_key: routing_key, payload: payload
      channel_connector.topic_exchange.publish(payload.to_json, publish_options.to_h)
      # if it gets this far it has succeeded
      logf "END Transmitting Message on id[%{id}] %{routing_key} -> %{payload}", id: publish_options.message_id, routing_key: routing_key, payload: payload
      true
    ensure
      channel_connector.connection_close if close_connector
    end
```

And then in *this* class we can:

```ruby
transmitter.send_message!(..., close_connector: false)
```

That's probably not 100% right, but these two methods are almost identical.

Reply:
If there is a failure halfway through (even with async error handling), there will still be the remainder of the batch that needs to be accounted for. Should this persistent transmitter apply the same async handling for the remainder? Otherwise, callers would need a way to track success per-message -- or resort to re-running the entire batch.